src/app/auth/signup.tsx

```diff
@@ -1,14 +1,99 @@
-import { View } from 'react-native';
-import Account from '../../components/Account';
-import Login from '../../components/Login';
+import React, { useEffect, useState } from 'react';
+import { Redirect, Link, router } from 'expo-router';
+import { Alert, StyleSheet, View, Text } from 'react-native';
+import { Button, Input } from 'react-native-elements';
 import { useSession } from '../../utils/AuthContext';
 
+const styles = StyleSheet.create({
+  container: {
+    marginTop: 40,
+    padding: 12,
+  },
+  verticallySpaced: {
+    paddingTop: 4,
+    paddingBottom: 4,
+    alignSelf: 'stretch',
+  },
+  mt20: {
+    marginTop: 20,
+  },
+});
+
 function SignUpScreen() {
-  const { session } = useSession();
+  const { session, signUp, signInWithEmail } = useSession();
+
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [signedUp, setSignedUp] = useState(false);
+
+  if (session) {
+    return <Redirect href="/home" />;
+  }
+
+  const signIn = async () => {
+    setLoading(true);
+    const { error } = await signInWithEmail(email, password);
+
+    if (error) Alert.alert(error.message);
+    else router.push('/auth/onboarding');
```

Comment:
I think for all of the nav we're doing in the auth flow we're going to want to explore using replace instead of push, that way users can't go back to a prior screen once they navigate to the next one. You can read more about that [here](https://docs.expo.dev/routing/navigating-pages/), and it'll be part of your next sprint task.